src/pages/federation.tsx

```diff
@@ -160,4 +243,202 @@ data.post("/refresh", async (c) => {
   return c.redirect("/federation?error=refresh");
 });
 
+data.post("/relay", async (c) => {
+  const fedCtx = federation.createContext(c.req.raw, undefined);
+  const form = await c.req.formData();
+
+  const inboxUrl = form.get("inbox_url");
+
+  if (typeof inboxUrl !== "string") {
+    return c.redirect("/federation?error=invalid_relay");
+  }
+
+  const [loadedRelay] = await db.transaction(async (tx) => {
+    await tx
+      .insert(instances)
+      .values({
+        host: fedCtx.host,
+        software: "hollo",
+        softwareVersion: null,
+      })
+      .onConflictDoNothing();
+
+    const existingRelay = await tx.query.relays.findFirst({
+      where: eq(relays.inboxUrl, inboxUrl),
+    });
+
+    if (existingRelay) {
+      throw tx.rollback();
+    }
+
+    // Create client actor for the relay if it doesn't exist
+    const account = await tx
+      .insert(accounts)
+      .values({
+        id: HOLLO_RELAY_ACTOR_ID,
+        iri: fedCtx.getActorUri(HOLLO_RELAY_ACTOR_USERNAME).href,
+        instanceHost: fedCtx.host,
+        type: "Application",
+        name: HOLLO_RELAY_ACTOR_USERNAME,
+        handle: `@${HOLLO_RELAY_ACTOR_USERNAME}@${fedCtx.host}`,
+        url: fedCtx.getActorUri(HOLLO_RELAY_ACTOR_USERNAME).href,
+        protected: false,
+        inboxUrl: fedCtx.getInboxUri(HOLLO_RELAY_ACTOR_USERNAME).href,
+        followersUrl: fedCtx.getFollowersUri(HOLLO_RELAY_ACTOR_USERNAME).href,
+        sharedInboxUrl: fedCtx.getInboxUri().href,
+        featuredUrl: fedCtx.getFeaturedUri(HOLLO_RELAY_ACTOR_USERNAME).href,
+        published: new Date(),
+      })
+      .onConflictDoUpdate({
+        target: accounts.id,
+        set: {
+          // Set anything to still return a value if the account already exists
+          protected: false,
+        },
+      })
+      .returning();
+
+    const rsaKeyPair = await generateCryptoKeyPair("RSASSA-PKCS1-v1_5");
+    const ed25519KeyPair = await generateCryptoKeyPair("Ed25519");
+
+    await tx
+      .insert(accountOwners)
+      .values({
+        id: HOLLO_RELAY_ACTOR_ID,
+        handle: HOLLO_RELAY_ACTOR_USERNAME,
+        rsaPrivateKeyJwk: await exportJwk(rsaKeyPair.privateKey),
+        rsaPublicKeyJwk: await exportJwk(rsaKeyPair.publicKey),
+        ed25519PrivateKeyJwk: await exportJwk(ed25519KeyPair.privateKey),
+        ed25519PublicKeyJwk: await exportJwk(ed25519KeyPair.publicKey),
+        bio: "This account is an internal account used by Hollo. It is used to follow remote relays.",
+        language: "en",
+        // TODO: Which visibility should be set?
+        visibility: "public",
+        discoverable: false,
+        themeColor: "pink",
+      })
+      .onConflictDoNothing()
+      .returning();
+
+    // Create follow request for this relay
+    const followRequestId = new URL(
+      `#relay-follows/${crypto.randomUUID()}`,
+      account[0].iri,
+    );
```

Comment:
Should this actually be a real Object ID, and not using the hash fragment?

Reply:
Perhaps we could use `followAccount` instead? Since that'll send the follow request and store it in the database.